rust/agama-network/src/dbus/service.rs

```diff
@@ -0,0 +1,37 @@
+//! Network D-Bus service.
+//!
+//! This module defines a D-Bus service which exposes Agama's network configuration.
+use crate::NetworkSystem;
+use agama_lib::connection_to;
+use std::{error::Error, thread};
+
+/// Represents the Agama networking D-Bus service.
+///
+/// It is responsible for starting the [NetworkSystem] on a different thread.
+pub struct NetworkService;
+
+impl NetworkService {
+    /// Starts listening and dispatching events on the D-Bus connection.
+    pub async fn start_service(address: &str) -> Result<(), Box<dyn Error>> {
+        let connection = connection_to(address).await?;
+        let mut network = NetworkSystem::from_network_manager(connection.clone())
```

Comment:
so this means that network manager has to use same DBus socket as agama one. Is the restriction needed? I though that for containerized one we have NM on different bus.

Reply:
Actually no, it uses a different connection (if you [check the internals](https://github.com/openSUSE/agama/blob/0d1defaaa29429dff9896d3c803bdcedbd7c0ff8/rust/agama-network/src/system.rs#L33-L35)). But it is true that it is not clear from this call. I could try to inject the adapter or to improve the docs to make it clear.